Changed the value of the parameter value of the Adam optimizer beta2 in

gwenniger opened the pull request:
onmt/Optim.py.
Change it from the value 0.98
(earlier adopted from "Attention is all you need"
https://arxiv.org/pdf/1706.03762.pdf)
to the more widely established value of 0.999
which was sugggested in the original paper about Adam
https://arxiv.org/pdf/1412.6980.pdf.
The new value of 0.999 is still being used by default in the main nmt architectures
and frameworks, including tensorflow and kerras.
https://www.tensorflow.org/api_docs/python/tf/train/AdamOptimizer
https://keras.io/optimizers/

	modified:   onmt/Optim.py

srush commented:
Oh, actually, let's make these command-line options as people may try different values. Also be sure your PR passes flake8. 

1.56s$ flake8
./onmt/Optim.py:21:80: E501 line too long (85 > 79 characters)
./onmt/Optim.py:25:35: W291 trailing whitespace
./onmt/Optim.py:26:80: E501 line too long (96 > 79 characters)
./onmt/Optim.py:27:80: E501 line too long (92 > 79 characters)
./onmt/Optim.py:28:80: E501 line too long (100 > 79 characters)
./onmt/Optim.py:28:101: W291 trailing whitespace
./onmt/Optim.py:32:1: E128 continuation line under-indented for visual indent

gwenniger commented:
Hi Sasha,
Thanks. I fixed everything and believe it
should be ok now.
I made a new pull request for the new
fixed version (2 commits).
By the way, my preleminary results do
indicate the new beta2 is really better,
quite a lot actually.

Gideon

On Tue, Nov 14, 2017 at 9:57 PM, srush <notifications@github.com> wrote:

> Oh, actually, let's make these command-line options as people may try
> different values. Also be sure your PR passes flake8.
>
> 1.56s$ flake8
> ./onmt/Optim.py:21:80: E501 line too long (85 > 79 characters)
> ./onmt/Optim.py:25:35: W291 trailing whitespace
> ./onmt/Optim.py:26:80: E501 line too long (96 > 79 characters)
> ./onmt/Optim.py:27:80: E501 line too long (92 > 79 characters)
> ./onmt/Optim.py:28:80: E501 line too long (100 > 79 characters)
> ./onmt/Optim.py:28:101: W291 trailing whitespace
> ./onmt/Optim.py:32:1: E128 continuation line under-indented for visual
> indent
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenNMT/OpenNMT-py/pull/383#issuecomment-344412607>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB6xzUydl7l9EPd-IS1kl_4qC35AhS1Eks5s2gy-gaJpZM4QeBRR>
> .
>

maksymbevza commented on opts.py:
```diff
@@ -195,6 +195,24 @@ def train_opts(parser):
+                        Typically a value of 0.999 is recommended, as this is
+                        the value suggested by the original paper describing
+                        Adam, and is also the value adopted in other frameworks
+                        such as Tensorflow and Kerras, i.e. see:
```
There's a misspelling in `Kerras`
